internal/core/src/segcore/InsertRecord.h

```diff
@@ -105,6 +114,8 @@ class OffsetOrderedMap : public OffsetMap {
             limit = map_.size();
         }
 
+        std::shared_lock<std::shared_mutex> lck(mtx_);
```

Comment:
should move to  line 113? since map_.size() should be protected.

Reply:
Yes.